src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx

```diff
@@ -30,31 +30,338 @@ const httpLink = new HttpLink({
+    });
+
+    waitFor(() => {
+      expect(screen.getByText('Install')).toBeInTheDocument();
+    });
+  });
+
+  test('modifies a plugin when "Modify" button is clicked', async () => {
+    render(
+      <ApolloProvider client={client}>
+        <Provider store={store}>
+          <BrowserRouter>
+            <I18nextProvider i18n={i18nForTest}>
+              <AddOnStore />
+            </I18nextProvider>
+          </BrowserRouter>
+        </Provider>
+      </ApolloProvider>
+    );
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      fireEvent.click(screen.getByText('Modify'));
+    });
+    waitFor(() => {
+      expect(screen.getByText('Modify')).toBeInTheDocument();
+    });
+  });
+
+  test('displays the organization screen with the correct title', async () => {
+    render(
+      <ApolloProvider client={client}>
+        <Provider store={store}>
+          <BrowserRouter>
+            <I18nextProvider i18n={i18nForTest}>
+              <AddOnStore />
+            </I18nextProvider>
+          </BrowserRouter>
+        </Provider>
+      </ApolloProvider>
+    );
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      expect(screen.getByTestId('organization-screen')).toBeInTheDocument();
+    });
+
+    waitFor(() => {
+      expect(screen.getByText('Plugin Store')).toBeInTheDocument();
+    });
+  });
+
+  test('handles API error and displays an error message', async () => {
+    jest.mock('@apollo/client', () => ({
+      ...jest.requireActual('@apollo/client'),
+      useQuery: jest.fn(() => ({
+        data: undefined,
+        loading: false,
+        error: new Error('API error'),
+      })),
+    }));
+
+    render(
+      <ApolloProvider client={client}>
+        <Provider store={store}>
+          <BrowserRouter>
+            <I18nextProvider i18n={i18nForTest}>
+              <AddOnStore />
+            </I18nextProvider>
+          </BrowserRouter>
+        </Provider>
+      </ApolloProvider>
+    );
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      expect(screen.getByText('Error loading data')).toBeInTheDocument();
+    });
+  });
+
+  test('clicks on an AddOnEntry and dispatches the correct Redux action', async () => {
+    render(
       <ApolloProvider client={client}>
         <Provider store={store}>
           <BrowserRouter>
             <I18nextProvider i18n={i18nForTest}>
-              {<AddOnStore />}
+              <AddOnStore />
             </I18nextProvider>
           </BrowserRouter>
         </Provider>
       </ApolloProvider>
     );
-    expect(getByTestId('AddOnEntryStore')).toBeInTheDocument();
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      fireEvent.click(screen.getByText('Example Plugin'));
+    });
+
+    waitFor(() => {
+      expect(screen.getByText('Example Plugin')).toBeInTheDocument();
```

Comment:
How does this imply "correct Redux action is dispatched"?

Reply:
it is working as it is expected to be

Reply:
See the test says `clicks on an AddOnEntry and dispatches the correct Redux action`, but you are testing if "Example Plugin" is present on the screen. How are these even related?

Reply:
oh I see let me see it in my code.